runtime/interpreter/value.go

```diff
@@ -18633,12 +18638,8 @@ func (v *DictionaryValue) IterateKeys(
 	locationRange LocationRange,
 	f func(key Value) (resume bool),
 ) {
-	valueComparator := newValueComparator(interpreter, locationRange)
-	hashInputProvider := newHashInputProvider(interpreter, locationRange)
 	iterate := func(fn atree.MapElementIterationFunc) error {
-		return v.dictionary.IterateKeys(
-			valueComparator,
-			hashInputProvider,
+		return v.dictionary.IterateReadOnlyKeys(
```

Comment:
Same here: Not sure that `f` in all uses of `DictionaryValue.IterateKeys` are always read-only

Reply:
Good catch! I reverted to use non-readonly iterator here.